sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -206,9 +206,9 @@ internal TimerCallback PerformExpiration()
                             {
                                 // At this point the pool item may have been closed already
                                 // if there was a context switch between the if conditions
-                                // and the pool item clean up kicked in.
+                                // and the pool item clean up started.
 
-                                poolItem.PartitionProducer.CloseAsync(CancellationToken.None).GetAwaiter().GetResult();
+                                poolItem.PartitionProducer.CloseAsync(false, CancellationToken.None);
```

Comment:
This may need to be wrapped in a set of pragmas to shut the analyzer up.   It's not a public synchronous method, so that guidance from the analyzer isn't valid here.

Example:
```csharp
#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
poolItem.PartitionProducer.CloseAsync(false, CancellationToken.None).GetAwaiter().GetResult();
#pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
```

Reply:
Out of curiosity, does it happen you to know why the analyzer suggests using ```EnsureCompleted()``` when ```EnsureCompleted()``` actually wraps a call to ```GetAwaiter().GetResult()```?

Reply:
I had similar questions.   This issue was opened to offer context:  https://github.com/Azure/azure-sdk-for-net/issues/10083

Reply:
thank you